File: WebAssemblyBrowserApp/WebAssembly_WebGPU/wwwroot/jsengine/core/storage.js
// =====================================================================
// 本地存储（localStorage，含异常兜底）。
// =====================================================================

export const storage = {
    get(key, fallback) {
        try {
            const v = localStorage.getItem(key)
            return v === null ? fallback : v
        } catch { return fallback }
    },
    set(key, value) {
        try { localStorage.setItem(key, value) } catch { /* 忽略 */ }
    },
}

File: WebAssemblyBrowserApp/WebEngineCommon/Engine/Storage.cs
using System.Runtime.InteropServices.JavaScript;

namespace WebAssemblyBrowserApp.Engine;

/// <summary>浏览器 localStorage 封装（JS 薄层），用于存档 / 最高分等。</summary>
public static partial class Storage
{
    [JSImport("storage.get", "main.js")]
    public static partial string Get(string key, string fallback);

    [JSImport("storage.set", "main.js")]
    public static partial void Set(string key, string value);
}
